File: UIexpenseTrackerWeb/src/app/services/income/income.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';

const BASIC_URL = "http://localhost:8080/api";

@Injectable({
  providedIn: 'root'
})
export class IncomeService {

  constructor(private http:HttpClient) { }

   postIncome(incomeDTO: any): Observable<any> {
   return this.http.post(BASIC_URL + "/income" , incomeDTO );
  }

  getAllIncomes(): Observable<any> {
     return this.http.get(BASIC_URL + "/income/all");
  }

  getIncomeById(id: number): Observable<any> {
     return this.http.get(BASIC_URL + `/income/${id}`);
  }

   updateIncome(id: number , incomeDTO : any): Observable<any> {
     return this.http.put(BASIC_URL + `/income/${id}`, incomeDTO);
  }

   deleteIncome(id: number): Observable<any> {
     return this.http.delete(BASIC_URL + `/income/${id}`);
  }

}

File: src/main/java/com/personal/ExpenseTracker/services/income/IncomeService.java
package com.personal.ExpenseTracker.services.income;

import com.personal.ExpenseTracker.dto.IncomeDTO;
import com.personal.ExpenseTracker.entity.Income;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public interface IncomeService {
    Income postIncome(IncomeDTO incomeDTO);

    List<Income> getAllIncomes();

    Income updateIncome(Long id, IncomeDTO incomeDTO);

    IncomeDTO getIncomeById(Long id);

    void deleteIncome(Long id);
}
